src/txn/txn_recover.c

```diff
@@ -688,6 +690,18 @@ __wt_txn_recover(WT_SESSION_IMPL *session)
 done:
     WT_ERR(__recovery_set_checkpoint_timestamp(&r));
 
+    /*
+     * Check whether the history store file exists or not. If it does not, then we should not apply
+     * rollback to stable to each table and we should not perform a checkpoint. This might happen if
+     * we're upgrading from an older version.
+     */
+    ret = __wt_hs_cursor(session, &session_flags, &is_owner);
+    if (ret == ENOENT) {
```

Comment:
Opening a metadata cursor and searching for the history store file in the metadata seems like the best method.

Look for:
if ((ret = session->open_cursor(session, WT_METADATA_URI, NULL, NULL, &cursor)) != 0) {

in util_list.c

Reply:
I tried this, however, if we go `4.4 => 4.2 => 4.4` we'll have the history store in metadata but it won't necessarily be on disk.

Reply:
1b7b5f8

Thanks, I tried this before but thought it didn't work. Seems fine.